crates/bevy_tasks/src/single_threaded_task_pool.rs

```diff
@@ -4,32 +4,26 @@ use core::{cell::RefCell, future::Future, marker::PhantomData, mem};
 
 use crate::Task;
 
-#[cfg(feature = "std")]
-use std::thread_local;
+crate::cfg::std! {
+    if {
+        use std::thread_local;
+        use crate::executor::LocalExecutor;
 
-#[cfg(not(feature = "std"))]
-use bevy_platform::sync::{Mutex, PoisonError};
+        thread_local! {
+            static LOCAL_EXECUTOR: LocalExecutor<'static> = const { LocalExecutor::new() };
+        }
 
-#[cfg(feature = "std")]
-use crate::executor::LocalExecutor;
+        type ScopeResult<T> = alloc::rc::Rc<RefCell<Option<T>>>;
+    } else {
+        use bevy_platform::sync::{Mutex, PoisonError};
+        use crate::executor::Executor as LocalExecutor;
 
-#[cfg(not(feature = "std"))]
-use crate::executor::Executor as LocalExecutor;
+        static LOCAL_EXECUTOR: LocalExecutor<'static> = const { LocalExecutor::new() };
 
-#[cfg(feature = "std")]
-thread_local! {
-    static LOCAL_EXECUTOR: LocalExecutor<'static> = const { LocalExecutor::new() };
+        type ScopeResult<T> = Arc<Mutex<Option<T>>>;
+    }
```

Comment:
I didn't know this syntax was supported. It's a bit weird but def an improvement.